### PR TITLE
add xgboost-pyspark demo

### PR DESCRIPTION
**Pull request by wbo4958**
This PR changed the XGBoost python examples based on dmlc/xgboost pyspark wrapper which will be included in the coming XGBoost 1.7 release.

I tested it locally on taxi/mortgage cpu/gpu runS and crossvalidator runS, and all of them can pass.

dmlc xgboost-pyspark has followed the xgboost naming way, just like below

``` bash
format=parquet
example=mortgage
main="main"

spark-submit \
  --master=local[1] \
  --py-files samples.zip \
  main.py \
  --mainClass com.nvidia.spark.examples.$example.$main\
  --dataPath=train::/data/unit_test_data/$example/$format/train \
  --dataPath=trans::/data/unit_test_data/$example/$format/trans \
  --mode=all \
  --format=$format \
  --max_depth=6 \
  --n_estimators=2 \
  --use_gpu=True \
```

So this PR is a breaking changing.

### CODE REVIEW

**Comment by wbo4958** [comment]
@GaryShen2008 @nvliyuan @NvTimLiu Hi, I verified this PR in the newest xgboost 1.7.0 rc. Could you help to review it?

Here is the testing env. 
``` bash
conda create -n cudf-pip python=3.9

conda activate cudf-pip
pip install cudf-cu11 --extra-index-url=https://pypi.ngc.nvidia.com
pip install sklearn
pip install xgboost==1.7.0rc1
```

**Comment by nvliyuan** [comment]
since xgboost 1.7.0 is not released yet, could we target this pr to v22.12 branch? we also need some time to update readmes and notebooks

**Comment by wbo4958** [comment]
@GaryShen2008 @nvliyuan @NvTimLiu , XGBoost has released the 1.7.0 officially and I verified the PR can run on xgboost 1.7.0 release. Please help to review again. Thx

**Comment by NvTimLiu** [comment]
I'll verify this PR in Jenkins jobs

**Comment by nvliyuan** [comment]
@wbo4958 why do we remove CPU codes?

File: examples/XGBoost-Examples/pack_pyspark_example.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Follow these steps to package the Python zip file
+rm -fr samples.zip
```
Comment (nvliyuan) [inline examples/XGBoost-Examples/pack_pyspark_example.sh:2]: we have a doc about how to build for python code [here](https://github.com/NVIDIA/spark-rapids-examples/blob/main/docs/get-started/xgboost-examples/building-sample-apps/python.md)

**Comment by nvliyuan** [comment]
the etl code failed with error:
```
Traceback (most recent call last):
  File "/home/yuali_nvidia_com/review/pr232/spark-rapids-examples/examples/XGBoost-Examples/main.py", line 16, in <module>
    from com.nvidia.spark.examples.main import main
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "/home/yuali_nvidia_com/review/pr232/spark-rapids-examples/examples/XGBoost-Examples/samples.zip/com/nvidia/spark/examples/main.py", line 16, in <module>
    from com.nvidia.spark.examples.main import main
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "/home/yuali_nvidia_com/review/pr232/spark-rapids-examples/examples/XGBoost-Examples/samples.zip/com/nvidia/spark/examples/utility/args.py", line 69, in <module>
TypeError: 'type' object is not subscriptable
```

**Comment by nvliyuan** [comment]
> the etl code failed with error:

same error while running xgboost demo, @NvTimLiu @wbo4958 could you help double check?

**Comment by wbo4958** [comment]
@nvliyuan Thx for the testing, yeah, the issue is the python typing compatibility issue. just had the commit to fix it. Thx

**Comment by nvliyuan** [comment]
@wbo4958 after fetching the new commit, got the below error:
```
Traceback (most recent call last):
  File "/home/yuali_nvidia_com/review/pr232/spark-rapids-examples/examples/XGBoost-Examples/main.py", line 18, in <module>
    main()
  File "/home/yuali_nvidia_com/review/pr232/spark-rapids-examples/examples/XGBoost-Examples/samples.zip/com/nvidia/spark/examples/main.py", line 21, in main
  File "/home/yuali_nvidia_com/review/pr232/spark-rapids-examples/examples/XGBoost-Examples/samples.zip/com/nvidia/spark/examples/utility/args.py", line 108, in parse_arguments
  File "/home/yuali_nvidia_com/review/pr232/spark-rapids-examples/examples/XGBoost-Examples/samples.zip/com/nvidia/spark/examples/utility/args.py", line 71, in _inspect_xgb_parameters
ModuleNotFoundError: No module named 'xgboost'
```
could you help check?

**Comment by NvTimLiu** [comment]
> 
I could not run xgboost work 

`FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/cuda/nvvm/lib64'`

**Comment by NvTimLiu** [comment]
Ran in nvidia/cuda:11.2-devel-ubuntu18.04 docker, still got below error @wbo4958  need we run under cuda11.5?

```
py4j.protocol.Py4JJavaError: An error occurred while calling z:org.apache.spark.api.python.PythonRDD.collectAndServe.
: org.apache.spark.SparkException: Job aborted due to stage failure: Could not recover from a failed barrier ResultStage. Most recent failure reason: Stage failed because barrier task ResultTask(3, 0) finished unsuccessfully.
org.apache.spark.api.python.PythonException: Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/cupy/__init__.py", line 18, in <module>
    from cupy import _core  # NOQA
  File "/opt/conda/lib/python3.9/site-packages/cupy/_core/__init__.py", line 1, in <module>
    from cupy._core import core  # NOQA
  File "cupy/_core/core.pyx", line 1, in init cupy._core.core
  File "cupy/_core/_kernel.pyx", line 7, in init cupy._core._kernel
  File "/opt/conda/lib/python3.9/site-packages/cupy/cuda/__init__.py", line 8, in <module>
    from cupy.cuda import compiler  # NOQA
  File "/opt/conda/lib/python3.9/site-packages/cupy/cuda/compiler.py", line 18, in <module>
    from cupy_backends.cuda.libs import nvrtc
ImportError: /opt/conda/lib/python3.9/site-packages/cupy_backends/cuda/libs/nvrtc.cpython-39-x86_64-linux-gnu.so: symbol nvrtcGetNVVMSize, version libnvrtc.so.11.2 not defined in file libnvrtc.so.11.2 with link time reference

```

**Comment by nvliyuan** [comment]
@NvTimLiu @wbo4958 I just make it works in local, but not worked in yarn due to "xgboost module not found" error, I suspect the root cause related to the wrong $PYSPARK_PYTHON, will verify it

**Comment by wbo4958** [comment]
After debug offline, the PR can run on dataproc on Yarn.

**Comment by nvliyuan** [comment]
@wbo4958 thanks for troubleshooting, work on dataproc on yarn environment

**Comment by nvliyuan** [comment]
approved since also works in docker @NvTimLiu

**Review by nvliyuan — APPROVED** [review]

**Comment by NvTimLiu** [comment]
verified in local docker container provided by@wbo4958, xgboost-pyspark test works now +1

**Review by NvTimLiu — DISMISSED** [review]

**Comment by nvliyuan** [comment]
@wbo4958 could you help target to 22.12 branch

**Comment by wbo4958** [comment]
@NvTimLiu,

I tried nvidia/cuda:11.2.2-devel-ubuntu18.04  and nvidia/cuda:11.5.1-devel-ubuntu18.04, both of them can work well.

These two docker images are lacking of python/java env, so first please install them by referencing https://docs.datastax.com/en/jdk-install/doc/jdk-install/installOpenJdkDeb.html https://stackoverflow.com/questions/60824700/how-to-install-python3-9-on-linux-ubuntu-terminal https://stackoverflow.com/questions/65644782/how-to-install-pip-for-python-3-9-on-ubuntu-20-04

after python java installed, 

``` bash
pip install cudf-cu11 --extra-index-url=https://pypi.ngc.nvidia.com
pip install xgboost
pip install scikit-learn
```

For cuda:11.2.2-devel-ubuntu18.04 images, need to re-install cupy by `pip uninstall cupy-cuda115; pip install cupy-cuda11x
`
please also refer to https://rapids.ai/pip.html#install

**Comment by wbo4958** [comment]
Re-target to 22.12

**Review by nvliyuan — APPROVED** [review]

**Pull request merged**